databuilder/extractor/bigquery_watermark_extractor.py

```diff
@@ -0,0 +1,140 @@
+from collections import namedtuple
+
+import logging
+import datetime
+
+from pyhocon import ConfigTree  # noqa: F401
+from typing import List, Any  # noqa: F401
+
+from databuilder.extractor.base_bigquery_extractor import BaseBigQueryExtractor
+from databuilder.models.watermark import Watermark
+
+DatasetRef = namedtuple('DatasetRef', ['datasetId', 'projectId'])
+TableKey = namedtuple('TableKey', ['schema_name', 'table_name'])
+PartitionInfo = namedtuple('PartitionInfo', ['partition_id', 'epoch_created'])
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BigQueryWatermarkExtractor(BaseBigQueryExtractor):
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+        BaseBigQueryExtractor.init(self, conf)
+        self.iter = iter(self._iterate_over_tables())
+
+    def get_scope(self):
+        # type: () -> str
+        return 'extractor.bigquery_watermarks'
+
+    def _is_sharded_table(self, table_id):
+        suffix = table_id[-8:]
+        return suffix.isdigit()
+
+    def _retrieve_tables(self, dataset):
+        # type: () -> Any
+        sharded_table_watermarks = {}
+
+        for page in self._page_table_list_results(dataset):
+            if 'tables' not in page:
+                continue
+
+            for table in page['tables']:
+                tableRef = table['tableReference']
+                table_id = tableRef['tableId']
+
+                if self._is_sharded_table(table_id):
+                    suffix = table_id[-8:]
```

Comment:
could you put a comment on how table_id looks like? 
also make a constant for 8